lib/_emerge/EbuildPhase.py

```diff
@@ -94,6 +96,31 @@ def _start(self):
         self._start_task(AsyncTaskFuture(future=future), self._async_start_exit)
 
     async def _async_start(self):
+
+        eapi_attrs = _get_eapi_attrs(self.settings["EAPI"])
+        if eapi_attrs.posixish_locale:
+            split_LC_ALL(self.settings)
```

Comment:
I've realized that this split_LC_ALL call will delete LC_ALL, and the config.reset() method will not restore the old LC_ALL unless it is in the config's backupenv. So, I need to think about how to fix this.

Reply:
I've solved this by making split_LC_ALL add an empty LC_ALL setting instead of deleting it. And when eapi_attrs.posixish_locale is True, config.environ() will assert correct LC_ALL state, and delete the empty LC_ALL setting.